libbeat/api/make_listener_windows.go

```diff
@@ -61,3 +63,24 @@ func makeListener(cfg Config) (net.Listener, error) {
 
 	return net.Listen(network, path)
 }
+
+func createListenerWithRetry(log *logp.Logger, pipe string, sd string) (net.Listener, error) {
```

Comment:
The point in the elastic-agent PR about this being uninterruptible, especially once we get to 30s, stands here as well.

This needs to take a context, it looks like there are not that many uses of this to update which is nice.